modules/drivers/mongo/src/metabase/driver/mongo/util.clj

```diff
@@ -89,35 +90,119 @@
     :else                         (throw (Exception. (str "with-mongo-connection failed: bad connection details:"
                                                           (:details database))))))
 
+(defn- srv-conn-str
+  "Creates Mongo client connection string to connect using
+   DNS + SRV discovery mechanism."
+  [user pass host authdb]
+  (format "mongodb+srv://%s:%s@%s/%s" user pass host authdb))
+
+(defn- normalize-details [details]
+  (let [{:keys [dbname host port user pass ssl authdb tunnel-host tunnel-user tunnel-pass additional-options]
+         :or   {port 27017, pass "", ssl false}} details
+        ;; ignore empty :user and :pass strings
+        user             (when (seq user)
+                           user)
+        pass             (when (seq pass)
+                           pass)
+        authdb           (if (seq authdb)
+                           authdb
+                           dbname)]
+    {:host               host
+     :port               port
+     :user               user
+     :authdb             authdb
+     :pass               pass
+     :dbname             dbname
+     :ssl                ssl
+     :additional-options additional-options}))
+
+(defn- srv-connection-info
+  "Connects to Mongo using DNS SRV.  Requires FQDN for `host` in the format
```

Comment:
Sorry to be nitpicky but could you update the docstrings to reflect what's actually happening? This functon is not actually connecting to anything anymore so it's confusing to still have docstrings that say the function connects to something.

Reply:
Fixed, good catch